src/card/authentication.js

```diff
@@ -0,0 +1,37 @@
+const { Validate, Auth, fetchWithHTTPErr } = require('../utils');
+const errors = require('../errors');
+
+function createAuthentication(data) {
+  return new Promise((resolve, reject) => {
+    try {
+      const compulsoryFields = ['amount', 'tokenID'];
+      const missingFields = Validate.requiredFields(compulsoryFields).validate(
+        data,
+      );
+      if (missingFields.length > 0) {
+        let message = Validate.missingFieldsErrMsg(missingFields);
+        reject({ status: 400, code: errors.API_VALIDATION_ERROR, message });
+      }
+
+      fetchWithHTTPErr(
+        `${this.API_ENDPOINT}/credit_card_tokens/${data.tokenID}/authentications`, // eslint-disable-line max-len
+        {
+          method: 'POST',
+          headers: {
```

Comment:
i'm thinking of putting this as `Xendit` or `Card` object function so that it can be retrieved from these other functions, e.g.:
`this.constructBasicHeader()`

do you think it's better this way? maybe it can also receive parameter on whether to use public key or private key

Reply:
not sure the best practice for this though

Reply:
Oh great suggestion (y) now that I think about it, this should be on the `Xendit` class level

Reply:
I just realised that this is a package-level (same as `Card`) util so I don't think it's necessary we move this auth creation fn under `Xendit` class. We can but it's more of a preference than best practice on this one. Furthermore, I have already designed the sub-packages to be injected with Xendit options rather than the whole Xendit object as I feel that such approach is more compositional